Marlin/i2cEncoder.cpp

```diff
@@ -0,0 +1,1173 @@
+                    // in order to correct for "error" but avoid correcting for noise and non skips
+                    // it must be > threshold and have a difference average of < 10 and be < 2000 steps
+                    if(abs(error) > threshold * planner.axis_steps_per_mm[encoderAxis] && diffSum < 10*(ERROR_ARRAY_SIZE-1) && abs(error) < 2000) { //Check for persistent error (skip)
+                      SERIAL_ECHO(axis_codes[get_axis()]);//Verbose JDK
+                      SERIAL_ECHO(" diffSum: ");
+                      SERIAL_ECHO(diffSum/(ERROR_ARRAY_SIZE-1));
+                      SERIAL_ECHO(" err detected: ");    //Verbose JDK                
+                      SERIAL_ECHO(error / planner.axis_steps_per_mm[encoderAxis]);
+                      SERIAL_ECHOLN("mm");
+                      SERIAL_ECHOLN("Correcting");
+                      thermalManager.babystepsTodo[encoderAxis] = -lround(error);
+                    }
+                  }
+                #else
+                  if(abs(error) > threshold * planner.axis_steps_per_mm[encoderAxis]) {
+                    //SERIAL_ECHOLN(error);
+                    //SERIAL_ECHOLN(position);
+                    //thermalManager.babystepsTodo[encoderAxis] -= STEPRATE * sgn(error);
+                    thermalManager.babystepsTodo[encoderAxis] = -lround(error/2);
+                  }
+                #endif
+/*              break;
+            case ECM_PLANNER:
+              if(abs(error) > threshold * planner.axis_steps_per_mm[encoderAxis]) {
+                double axisPosition[NUM_AXIS];
+
+                for(int i = 0; i < NUM_AXIS; i++) {
+                  axisPosition[i] = stepper.get_axis_position_mm((AxisEnum)i);
+                }
+
+                axisPosition[get_axis()] = mm_from_count(position);
+                planner.set_position_mm(axisPosition[X_AXIS],axisPosition[Y_AXIS],axisPosition[Z_AXIS],axisPosition[E_AXIS]);
+                current_position[get_axis()] = mm_from_count(position);
+              }
+              break;
+            case ECM_STALLDETECT:
+
+              //if position unchanged from last several readings
+              //and stepper position should have changed
+              //sound the alert
+              break;
+
+          }
+*/
+
+          if(abs(error) > (ERROR_COUNTER_TRIGGER_THRESHOLD * planner.axis_steps_per_mm[encoderAxis]) && millis() - lastErrorCountTime > ERROR_COUNTER_DEBOUNCE_MS) {
+/*            SERIAL_ECHO("Large error on ");
+            SERIAL_ECHO(axis_codes[get_axis()]);
+            SERIAL_ECHO(" axis. error: ");
+            SERIAL_ECHO((int)error);
+            SERIAL_ECHO("; diffSum: ");
+            SERIAL_ECHOLN(diffSum);
+ */           errorCount++;
+            lastErrorCountTime = millis();
+          }
+        }
+
+        lastPositionTime = positionTime;
+      }/* else {
+
+        //if the magnetic strength has been good for a certain time, start trusting the module again
```

Comment:
With the right setup there are no problems, but this was included in the early code to allow for and manage the possibility that the magnet looses position because it is out of range of the magnetic strip during operation. In my experience this added more issues that what it may have resolved. If the magnetic strip looses position during the print in the current code my thoughts are that the encoders should get disabled and the user should be notified that this is happening by lcd message and serial. @chrisbarr thoughts?